### DISCUSSION TITLE
Mutual exclusion?

### DISCUSSION THREAD

**Discussion by tombohub** [bilal-fazlani/commanddotnet#471] [Q&A] [answered]
Hello,
does library support mutual exclusion? 
https://docs.python.org/3/library/argparse.html#mutual-exclusion

**Comment by drewburlingame** [comment, marked as answer]
Not declaratively, but we do support DataAnnotations validations and you can use an IArgumentModel that implements IValidatableObject to implement it, as shown in [this example](https://commanddotnet.bilal-fazlani.com/argumentvalidation/data-annotations-validation/#example)

**Reply by tombohub** [reply]
Ok, thank you

**Reply by drewburlingame** [reply]
@tombohub no problem.  I'm curious if this meets your need?  When I thought about adding mutual exclusivity, I realized personally that I preferred the flexibility provided by IValidatableObject to give a more descriptive error message and having exceptional cases where options can be mutually exclusive depending on another option.  I recognize that may not be everyone's experience.  I'm curious what your opinion is.

**Reply by tombohub** [reply]
Hi, for me personally I prefer to have quick way to mutually exclude since I'm hobby developer making tools for my personal needs. I'm not sure what would I prefer if it's more than that. In this case I'll probably do a quick validation cause I'm using method style. Do I need to use class style to use to use `IValidatableObject`?

**Reply by drewburlingame** [reply]
The command is still defined as a method.  The parameters would need to be in an `IArgumentModel`, as shown in the example.  That argument model class becomes a parameter on the method.
